lib/ae_mdw/db/sync/contract.ex

```diff
@@ -180,4 +184,31 @@ defmodule AeMdw.Db.Sync.Contract do
         |> NameRevokeMutation.new(call_txi, block_index)
     end)
   end
+
+  # Temporary HACK to fix oracle query nonces.
+  defp fix_oracle_queries(events, block_hash) do
+    {fixed_events, _acc} =
+      Enum.map_reduce(events, %{}, fn
+        {"Oracle.query", aetx}, accounts_nonces ->
+          {:oracle_query_tx, tx} = :aetx.specialize_type(aetx)
+          sender_pk = :aeo_query_tx.sender_pubkey(tx)
+
+          nonce =
+            case Map.fetch(accounts_nonces, sender_pk) do
+              {:ok, nonce} -> nonce
+              :error -> Db.nonce_at_block(block_hash, sender_pk)
+            end
+
+          accounts_nonces = Map.put(accounts_nonces, sender_pk, nonce + 1)
```

Comment:
The query is calculated based on `(oracle_pk, sender_pk, nonce)`, so if two `Oracle.query` happen on the same contract call the nonce needs to increment, and it does

Reply:
Please bear in mind that `nonces` are incremented only once per `:contract_call_tx` (it's not incremented by internal call).

Reply:
Sorry I didn't know there is a part of Node code that increments internally it also for internal calls. I am talking based on this activities:
https://testnet.aeternity.art/mdw/v2/accounts/ak_s6TEb7Kuw17DNMoak7dHmby85zUeGmYzvczUCjWLQk3m1ZebS/activities

where the second has two `Oracle.query` but the nonce is incremented only once.

Used this contract to confirm:
```
@compiler >= 6

contract OracleDelegated =
  record state = {oracles: map(address, address)}
  
  entrypoint init() =
    {oracles={}}

  payable stateful entrypoint createQueries(o : oracle(string, string), qfee : int, qttl : Chain.ttl, rttl : Chain.ttl) : oracle_query(string, string) =
    let q1 = "{'city': 'Paris'}"
    let q2 = "{'city': 'London'}"
    require(qfee =< Call.value, "insufficient value for qfee")
    require(Oracle.check(o), "oracle not valid")
    Oracle.query(o, q1, qfee, qttl, rttl)
    Oracle.query(o, q2, qfee, qttl, rttl)
```
but as you've shown the Node might artificially increment it for allowing multiple queries on internal calls to have unique oracle ids composed by `(oracle pk, sender pk, nonce)` 

If so it would need to handle only other cases where it requires some additional increments:
b) when there are other block/outer transactions from same sender before an internal call (e.g T1: name claim, T2: contract call with Oracle query should consider the nonce after T1)
c) when there are other block/outer transactions from same sender between internal calls (e.g T1: contract call Oracle query, T2: name update, T3: contract call Oracle query should consider the nonce after T1 and T2)